operator/informer_controller.go

```diff
@@ -239,20 +254,128 @@ func (c *InformerController) AddInformer(informer Informer, resourceKind string)
 		return fmt.Errorf("resourceKind cannot be empty")
 	}
 
-	err := informer.AddEventHandler(&SimpleWatcher{
-		AddFunc:    c.informerAddFunc(resourceKind),
-		UpdateFunc: c.informerUpdateFunc(resourceKind),
-		DeleteFunc: c.informerDeleteFunc(resourceKind),
-	})
+	watcher := NewConcurrentWatcher(
```

Comment:
We may want to only use the ConcurrentWatcher if `c.informerMaxWorkers > 1`, so we don't go through additional logic if the app doesn't need/want concurrency.

Reply:
Good point. It would also help us avoid some memory overhead of an additional buffer too. Plus cleaner spans, simpler logic.

Will do that and also add tests for the `ConcurrentWatcher`.